### DISCUSSION TITLE
insmod mqnic.ko failed-module verification failed: signature and/or required key missing - tainting kernel

### DISCUSSION THREAD

**Discussion by Xuhd08** [corundum/corundum#123] [Q&A]
Hi,
When I use the kernel module mqnic.ko, use dmesg command will show the module verification failed, I also used some methods such as self-signed the module, but all doesn't work.  
My os info： ubuntu 18.04 LTS-bionic ,and the kernel version is 4.15.0-192-generic

<img width="479" alt="image" src="https://user-images.githubusercontent.com/95950347/198813217-3315782c-9943-460a-95db-1da9faf516c0.png">

Could you help to check it and how to sovle the problem?

Thanks！

**Comment by alexforencich** [comment]
You need to disable signature verification.  

One method is to adjust the bootloader to add `module.sig_enforce=0` as a kernel parameter (see https://askubuntu.com/questions/773734/how-to-install-module-ko-module-without-kernel-signature-or-kernel-rebuild-in-ub).  You will also have to disable secure boot, if it is enabled.

**Reply by Xuhd08** [reply]
Hi，
I have add the parameter **module.sig_enforce=0** and the secure boot is disabled. but it still doesn,t work. 
![image](https://user-images.githubusercontent.com/95950347/198924434-1026146b-ccfb-4932-9bca-f4c3b3368ad1.png)
any other methods?

Thanks!

**Reply by Xuhd08** [reply]
Hi，
I have solved this issue. 
1, make sure the secure boot is disabled
2, cd /path/to/corundum/modules/mqnic
3, vi Makefile and add  `CONFIG_MODULE_SIG=n` after the command line "KDIR ?= /lib/modules/$(shell uname -r)/build"
4, save and quit vi, make again and loading the module , it worked for me

Thanks.
